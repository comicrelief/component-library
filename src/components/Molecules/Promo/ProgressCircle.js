import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import {
  ProgressRingWrapper, ProgressRingSVG, ProgressRingCircle
} from './ProgressCircle.style';

const ProgressCircle = ({ thisStroke, thisRadius, thisProgress }) => {
  const initNormRadius = thisRadius - thisStroke * 2;
  const thisCircumference = initNormRadius * 2 * Math.PI;
  const [thisDashOffset, setThisDashOffset] = useState(initNormRadius * 2 * Math.PI);

  useEffect(() => {
    const offset = thisCircumference - ((thisProgress / 100) * thisCircumference);
    setThisDashOffset(offset);
  }, [thisProgress, thisCircumference]);

  return (
    <ProgressRingWrapper thisProgress={thisProgress}>
      <ProgressRingSVG height={thisRadius * 2} width={thisRadius * 2}>
        <ProgressRingCircle
          strokeDasharray={`${thisCircumference} ${thisCircumference}`}
          strokeDashOffsetStyle={thisDashOffset}
          strokeWidth={`${thisStroke}`}
          r={`${initNormRadius}`}
          cx={`${thisRadius}`}
          cy={`${thisRadius}`}
        />
      </ProgressRingSVG>
    </ProgressRingWrapper>
  );
};

ProgressCircle.propTypes = {
  thisStroke: PropTypes.number.isRequired,
  thisRadius: PropTypes.number.isRequired,
  thisProgress: PropTypes.number.isRequired
};

export default ProgressCircle;
