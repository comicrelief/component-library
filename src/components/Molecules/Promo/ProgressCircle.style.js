import styled, { keyframes } from 'styled-components';
// import { media, container } from '../../../theme/shared/size';
// import spacing from '../../../theme/shared/spacing';
// import zIndex from '../../../theme/shared/zIndex';

const animCircle = keyframes`
    to {
        stroke-dashoffset: ${({ thisOffset }) => thisOffset};
    }
`;

const ProgressBar = styled.div`
    box-sizing: border-box;
    background-color: green;
    position: relative;
    width: 170px;
    height: 170px;
    margin: 1em;
    transform: rotate(-90deg);

    position: absolute;
    top: 0;
    right: 0;

    transition: transform 0.5s;
    transform: rotate( ${({ degreesLeft }) => degreesLeft}deg);
    width: 100px;
    height: 100px;
    border-radius: 50%;
    // position: relative;
    background: radial-gradient(#EF6868 0 9.5px, transparent 10.5px) 50% 0% / 20px 20px no-repeat, 
                radial-gradient(transparent 29px, #2493AB 30px 50px) 0 0 / 100% 100% no-repeat;
    animation: animate 3s linear infinite;

`;

const ProgressBarSvg = styled.svg`
    box-sizing: border-box;
    position: relative;
    width: 175px;
    height: 175px;
    overflow: hidden;
    display: none;
`;

const ProgresBarSvgCircle = styled.circle`
    box-sizing: border-box;
    width: 100%;
    height: 100%;
    fill: none;
    stroke-width: 10;
    stroke-dasharray: ${({ thisOffset }) => thisOffset};
    stroke-dashoffset: 0;
    stroke: red;
    stroke-linecap: round;
    transform: translate(5px, 5px); // stroke-width / 2
    // animation: ${animCircle} 1s ease-in-out forwards;

    display: none;
`;

export {
  ProgressBar, ProgressBarSvg, ProgresBarSvgCircle
};
