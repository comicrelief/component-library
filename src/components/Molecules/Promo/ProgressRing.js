import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import {
  ProgressRingWrapper, ProgressRingSVG, ProgressRingCircle
} from './PromoVideoButton.style';

const ProgressRing = ({
  thisStroke, thisRadius, videoProgress, isPlaying
}) => {
  const initNormRadius = thisRadius - thisStroke * 2;
  const thisCircumference = initNormRadius * 2 * Math.PI;
  const [thisDashOffset, setThisDashOffset] = useState(initNormRadius * 2 * Math.PI);

  useEffect(() => {
    const offset = thisCircumference - ((videoProgress / 100) * thisCircumference);
    setThisDashOffset(offset);
  }, [videoProgress, thisCircumference]);

  return (
    <ProgressRingWrapper videoProgress={videoProgress}>
      <ProgressRingSVG height={thisRadius * 2} width={thisRadius * 2}>
        <ProgressRingCircle
          strokeDasharray={`${thisCircumference} ${thisCircumference}`}
          strokeDashOffsetStyle={thisDashOffset}
          strokeWidth={`${thisStroke}`}
          r={`${initNormRadius}`}
          cx={`${thisRadius}`}
          cy={`${thisRadius}`}
          isPlaying={isPlaying}
        />
      </ProgressRingSVG>
    </ProgressRingWrapper>
  );
};

ProgressRing.propTypes = {
  thisStroke: PropTypes.number.isRequired,
  thisRadius: PropTypes.number.isRequired,
  videoProgress: PropTypes.number.isRequired,
  isPlaying: PropTypes.bool.isRequired
};

export default ProgressRing;
