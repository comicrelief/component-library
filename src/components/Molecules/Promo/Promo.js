import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import PromoVideo from './_PromoVideo';
import { sizes } from '../../../theme/shared/breakpoint';

import Picture from '../../Atoms/Picture/Picture';

import {
  Container, Wrapper, Copy, Media
} from './Promo.style';

const Promo = ({
  copyLeft,
  backgroundColor,
  imageLow,
  imageSet,
  image,
  imageAltText,
  children,
  position,
  autoPlay,
  loop,
  poster,
  mobilePoster,
  showPosterAfterPlaying,
  videoSrc,
  mobileVideoSrc,
  lightVideo
}) => {
  // Store the appropriate prop in state, dependent on the breakpoint
  const [thisVideoSrc, setThisVideoSrc] = useState(null);
  const [thisPoster, setThisPoster] = useState(null);

  // Let either field define this, just in case
  const hasVideo = Boolean(videoSrc || mobileVideoSrc);
  // A 'video-y' Promo will override and ignore any 'non-Video' images;
  // none of these fields are required so we have to handle them accordingly:
  const hasImage = Boolean(imageSet && !hasVideo);

  // Runs on initial load:
  useEffect(() => {
    if (hasVideo) {
      // Checks size on load ONLY; don't want to mess about with listeners:
      const isDesktop = window.innerWidth >= sizes.nav; // 1024+ px
      let currentPoster; let currentSrc;

      // If we've got both desktop AND mobile videos, let the
      // *current* screen width define which video src to use:
      if (videoSrc && mobileVideoSrc) {
        currentSrc = isDesktop ? videoSrc : mobileVideoSrc;
        currentPoster = isDesktop ? poster : mobilePoster;
      } else {
        // Else, pick whatever we DO actually have:
        currentSrc = videoSrc || mobileVideoSrc;
        currentPoster = poster || mobilePoster;
      }

      setThisVideoSrc(currentSrc);
      setThisPoster(currentPoster);
    }
  }, [hasVideo, videoSrc, mobileVideoSrc, mobilePoster, poster]);

  return (
    <Container backgroundColor={backgroundColor} position={position}>
      <Media imageRight={copyLeft}>
        {(hasImage && !hasVideo) && (
        <Picture
          alt={imageAltText}
          imageLow={imageLow}
          images={imageSet}
          image={image}
          objectFit="cover"
          width="100%"
          height="100%"
        />
        )}
        {/* If no poster image is supplied, it'll gracefully
        fall back to the first frame, so only checking for a src here */}
        {(hasVideo && thisVideoSrc) && (
        <>
          <PromoVideo
            copyLeft={copyLeft}
            lightVideo={lightVideo}
            thisVideoSrc={thisVideoSrc}
            thisPoster={thisPoster}
            showPosterAfterPlaying={showPosterAfterPlaying}
            autoPlay={autoPlay}
            loop={loop}
          />
        </>
        )}
      </Media>

      <Wrapper copyLeft={copyLeft}>
        <Copy position={position} hasVideo={hasVideo} lightVideo={lightVideo}>
          {children}
        </Copy>
      </Wrapper>
    </Container>
  );
};

Promo.propTypes = {
  backgroundColor: PropTypes.string,
  copyLeft: PropTypes.bool,
  imageLow: PropTypes.string,
  imageSet: PropTypes.string,
  image: PropTypes.string,
  imageAltText: PropTypes.string,
  children: PropTypes.node,
  position: PropTypes.oneOf(['upper', 'lower', 'end', 'none']),
  autoPlay: PropTypes.bool,
  loop: PropTypes.bool,
  videoSrc: PropTypes.string,
  mobileVideoSrc: PropTypes.string,
  poster: PropTypes.string,
  mobilePoster: PropTypes.string,
  showPosterAfterPlaying: PropTypes.bool,
  lightVideo: PropTypes.bool
};

Promo.defaultProps = {
  backgroundColor: 'white',
  copyLeft: false,
  imageSet: null,
  imageLow: null,
  image: null,
  imageAltText: '',
  children: null,
  position: 'none',
  autoPlay: true,
  loop: true,
  poster: null,
  mobilePoster: null,
  videoSrc: null,
  mobileVideoSrc: null,
  showPosterAfterPlaying: true,
  lightVideo: false
};

export default Promo;
