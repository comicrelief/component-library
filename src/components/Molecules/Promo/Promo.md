Single Message

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';

<Promo
  backgroundColor="blue_dark"
  imageSet={data.promoImage}
  image={data.promoImage}
  imageAltText=""
  copyFirst={false}
  nudge="bottom"
>
  <Text
    tag="h1"
    color="white"
    size="super"
    family="Anton"
    uppercase
    weight="normal"
  >
    Curabitur pretium tincidunt lacus
  </Text>
  <Text tag="p" color="white">
    Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius,
    turpis et commodo, est eros bibendum elit, nec luctus magna sollicitudin.
  </Text>
  <span>
    <Link href="/" type="button" color="white">
      Show me this
    </Link>
  </span>
</Promo>;
```
