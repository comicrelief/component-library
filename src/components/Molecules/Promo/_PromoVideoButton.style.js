import styled, { css } from 'styled-components';
import playWhite from './assets/Play--white.svg';
import pauseWhite from './assets/Pause--white.svg';
import playBlack from './assets/Play--black.svg';
import pauseBlack from './assets/Pause--black.svg';

const PlayButtonWrapper = styled.div`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  &:after {
    // Preload all button icons
    content: url(${playWhite}) url(${pauseWhite}) url(${playBlack}) url(${pauseBlack}) none;
  }
`;

const PlayButton = styled.button`
  display: flex;
  width: 50px;
  height: 50px;
  position: absolute;
  top: 10px;
  right: 10px;
  align-items: center;
  justify-content: center;
  border: 0;
  padding: 0;
  margin: 0;
  border-radius: 50%;
  background: ${({ theme }) => theme.color('white')};
  opacity: 0.8;
  transition: opacity 0.2s;

  &:hover,
  &:focus {
    opacity: 1.0;
  }

  &:focus-visible {
    outline-style: outset;
    outline-width: medium;
  }

  ${({ blackPlayButton }) => blackPlayButton && css`
    background: ${({ theme }) => theme.color('black')};
  `}

  ${({ copyLeft }) => !copyLeft && css`
    right: auto;
    left: 10px;
  `}

  @media ${({ theme }) => theme.breakpoint('medium')} {
    top: 20px;
    right: 20px;

      ${({ copyLeft }) => !copyLeft && css`
      right: auto;
      left: 20px;
    `}
  }
`;

const ProgressRingWrapper = styled.span`
    position: absolute;
    width: 100%;
    height: 100%;
    display: flex;
`;

const ProgressRingSVG = styled.svg`
 //
`;

const ProgressRingCircle = styled.circle`
    z-index: 100;
    transition: stroke-dashoffset 0.5s;

    transform: rotate(-90deg);
    transform-origin: 50% 50%;
    stroke-dashoffset: ${({ strokeDashOffsetStyle }) => strokeDashOffsetStyle};
    stroke: white; // need to changed based on Promo settings
    fill: transparent;

    // White button, black stroke
    ${({ blackPlayButton }) => !blackPlayButton && css`
      stroke: ${({ theme }) => theme.color('black')};
    `}

    // Cancel the animation
    ${({ isRestarting }) => (isRestarting) && css`
     transition: none;
    `}
`;

const Icon = styled.span`
  height: 50px;
  width: 50px;
  background: no-repeat center/50% url(${playWhite}) transparent;

  // Black button, white icons
  ${({ blackPlayButton, isPlaying }) => (blackPlayButton && !isPlaying) && css`
    background-image: url(${playWhite});
  `}

  ${({ blackPlayButton, isPlaying }) => (blackPlayButton && isPlaying) && css`
    background-image: url(${pauseWhite});
  `}

  // White button, black icons
  ${({ blackPlayButton, isPlaying }) => (!blackPlayButton && !isPlaying) && css`
    background-image: url(${playBlack});
  `}

  ${({ blackPlayButton, isPlaying }) => (!blackPlayButton && isPlaying) && css`
    background-image: url(${pauseBlack});
  `}
`;

export {
  ProgressRingWrapper, ProgressRingSVG, ProgressRingCircle, PlayButton, PlayButtonWrapper, Icon
};
