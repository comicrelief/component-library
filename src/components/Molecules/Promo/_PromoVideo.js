import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { Video } from './Promo.style';
import PromoVideoButton from './_PromoVideoButton';

const PromoVideo = ({
  behindTextGradient,
  copyLeft,
  thisVideoSrc,
  thisPoster,
  autoPlay,
  loop,
  showPosterAfterPlaying,
  blackPlayButton
}) => {
  const [isPlaying, setIsPlaying] = useState(false);
  const [isRestarting, setIsRestarting] = useState(false);
  const [videoProgress, setVideoProgress] = useState(0);
  const videoEl = useRef(null);

  const togglePlay = () => {
    if (isPlaying) videoEl.current.pause();
    else videoEl.current.play();
    setIsPlaying(!isPlaying);
  };

  const updateTime = () => {
    if (videoEl.current && videoEl.current.duration) {
      // Calculate the percentage of the video played:
      const percentage = Math.round((videoEl.current.currentTime / videoEl.current.duration) * 100);
      // Because a completely smooth animation is impossible with the 250ms-ish frequency of
      // the 'timeupdate' event, we're electing to just update the ring every 25%.
      const nearest = 25;
      const roundedPercentage = (percentage + nearest / 2) - ((percentage + nearest / 2) % nearest);
      setVideoProgress(roundedPercentage);
    }
  };

  // Only loads once the initial screensize check is complete
  useEffect(() => {
    // Use truthy comparison so either a null OR undefined value won't work
    if (thisVideoSrc != null) {
      videoEl.current.addEventListener('timeupdate', updateTime);
      // Trigger on-load autoplay if apppropriate
      if (autoPlay && !isPlaying) {
        togglePlay();
      }

      videoEl.current.addEventListener('ended', () => {
        // Used purely to halt the CSS animation:
        setIsRestarting(true);

        // If this is a non-looping video, add a listener to update our local state
        // once the video's ended, to let the user retrigger it manually:
        if (!loop) {
          setIsPlaying(false);
          setVideoProgress(0);
          // Reload the video to show the poster image:
          if (showPosterAfterPlaying) videoEl.current.load();
        } else {
          // Rather than using the Video 'loop' property, we retrigger
          // it in *code* as there's no 'restarted' to hook into:
          togglePlay();
        }

        // Grace period to allow the animation to reset
        setTimeout(() => { setIsRestarting(false); }, 100);
      });
    }
    // CERTAINLY don't want this re-running for EACH of these variable updates, sorry Lint...
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [thisVideoSrc]);

  return (
    <>
      <Video
        ref={videoEl}
        src={thisVideoSrc}
        poster={thisPoster}
        muted
      >
        Your browser does not support this video filetype.
      </Video>
      <PromoVideoButton
        copyLeft={copyLeft}
        videoProgress={videoProgress}
        togglePlay={togglePlay}
        isPlaying={isPlaying}
        behindTextGradient={behindTextGradient}
        blackPlayButton={blackPlayButton}
        isRestarting={isRestarting}
      />
    </>
  );
};

PromoVideo.propTypes = {
  copyLeft: PropTypes.bool.isRequired,
  behindTextGradient: PropTypes.oneOf(['black', 'white', 'none']).isRequired,
  blackPlayButton: PropTypes.bool.isRequired,
  thisVideoSrc: PropTypes.string.isRequired,
  thisPoster: PropTypes.string.isRequired,
  autoPlay: PropTypes.bool.isRequired,
  loop: PropTypes.bool.isRequired,
  showPosterAfterPlaying: PropTypes.bool.isRequired
};

export default PromoVideo;
