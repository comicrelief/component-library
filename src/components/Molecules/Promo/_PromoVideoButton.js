import React from 'react';
import PropTypes from 'prop-types';
import ProgressRing from './_ProgressRing';
import { PlayButtonWrapper, PlayButton, Icon } from './_PromoVideoButton.style';

const PromoVideoButton = ({
  togglePlay, ...rest
}) => (
  <PlayButtonWrapper className="weewee">
    <PlayButton
      onClick={() => { togglePlay(); }}
      {...rest}
    >
      <Icon {...rest} />
      <ProgressRing
        {...rest}
      />
    </PlayButton>
  </PlayButtonWrapper>
);

PromoVideoButton.propTypes = {
  togglePlay: PropTypes.func.isRequired
};

export default PromoVideoButton;
