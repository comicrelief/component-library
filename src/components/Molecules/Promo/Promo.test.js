import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import Promo from './Promo';
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';

const data = require('../../../styleguide/data/data').default;

it('renders Promo correctly', () => {
  const tree = renderWithTheme(
    <Promo
      backgroundColor="blue_dark"
      imageSet={data.promoImage}
      image={data.promoImage}
      imageAltText=""
      copyFirst={false}
      hasOverlay
    >
      <Text
        tag="h1"
        color="white"
        size="super"
        family="Anton"
        uppercase
        weight="normal"
      >
        Curabitur pretium tincidunt lacus
      </Text>
      <Text tag="p" color="white">
        Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius,
        turpis et commodo, est eros bibendum elit, nec luctus magna sollicitudin.
      </Text>
      <span>
        <Link href="/" type="button" color="white">
          Show me this
        </Link>
      </span>
    </Promo>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
