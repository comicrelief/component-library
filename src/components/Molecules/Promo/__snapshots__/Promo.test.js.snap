// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Promo correctly 1`] = `
.c3 {
  color: #FFFFFF;
  font-size: 5rem;
  line-height: 5rem;
  text-transform: uppercase;
  font-weight: normal;
  font-family: 'Anton',Impact,sans-serif;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
  font-size: 3rem;
  line-height: 3rem;
}

.c4 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  padding: 0.5rem 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 1rem;
  border-radius: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  height: 3.125rem;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #FFFFFF;
  color: #000000;
}

.c5:hover {
  background-color: #969598;
  color: #000000;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  background: #274084;
}

.c1 {
  width: 100%;
  max-width: 1200px;
  height: 100%;
  left: 0;
  right: 0;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  width: 100%;
  padding: 1.5rem 1.5rem 4rem;
  z-index: 1;
}

@media (min-width:740px) {
  .c3 {
    font-size: 4rem;
    line-height: 4rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 5rem;
    line-height: 5rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:740px) {
  .c5 {
    width: auto;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:1024px) {
  .c1 {
    min-height: calc(100vh - 90px);
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 70%;
    padding: 4rem 1.5rem;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 50%;
    padding: 8rem 1.5rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1
        className="c3"
        color="white"
        size="super"
      >
        Curabitur pretium tincidunt lacus
      </h1>
      <p
        className="c4"
        color="white"
        size="s"
      >
        Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius, turpis et commodo, est eros bibendum elit, nec luctus magna sollicitudin.
      </p>
      <span>
        <a
          className="c5"
          color="white"
          href="/"
          target="_self"
          type="button"
        >
          Show me this
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`renders Promo correctly end position 1`] = `
.c6 {
  color: #FFFFFF;
  font-size: 5rem;
  line-height: 5rem;
  text-transform: uppercase;
  font-weight: normal;
  font-family: 'Anton',Impact,sans-serif;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
  font-size: 3rem;
  line-height: 3rem;
}

.c7 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c2 {
  display: block;
  width: 100%;
  height: 100%;
  position: relative;
}

.c3 {
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  padding: 0.5rem 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  font-size: 1rem;
  border-radius: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  height: 3.125rem;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #FFFFFF;
  color: #000000;
}

.c8:hover {
  background-color: #969598;
  color: #000000;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  background: #274084;
  -webkit-clip-path: polygon(0 0,100% 0,100% 90%,0 101%);
  clip-path: polygon(0 0,100% 0,100% 90%,0 101%);
  border-radius: 0 0 0 2rem;
}

.c4 {
  width: 100%;
  max-width: 1200px;
  height: 100%;
  left: 0;
  right: 0;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  width: 100%;
  padding: 1.5rem 1.5rem 4rem;
  z-index: 1;
}

.c1 {
  width: 100%;
  height: auto;
}

.c1 img {
  object-position: center;
}

@media (min-width:740px) {
  .c6 {
    font-size: 4rem;
    line-height: 4rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:1024px) {
  .c6 {
    font-size: 5rem;
    line-height: 5rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:740px) {
  .c8 {
    width: auto;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-clip-path: polygon(0 0,100% 0,100% 85%,0% 101%);
    clip-path: polygon(0 0,100% 0,100% 85%,0% 101%);
    border-radius: 0 0 0 4rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    min-height: calc(100vh - 90px);
  }
}

@media (min-width:1024px) {
  .c5 {
    width: 70%;
    padding: 4rem 1.5rem;
  }
}

@media (min-width:1024px) {
  .c5 {
    width: 50%;
    padding: 8rem 1.5rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    height: 100%;
    position: absolute;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 lazyload"
      height="100%"
      width="100%"
    >
      <img
        alt=""
        className="c3 lazyload"
        data-lowsrc={null}
        data-sizes="auto"
        data-src="../promo.jpg"
        data-srcset="../promo.jpg"
        height="100%"
        src="../promo.jpg"
        srcSet="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        width="100%"
      />
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <h1
        className="c6"
        color="white"
        size="super"
      >
        Curabitur pretium tincidunt lacus
      </h1>
      <p
        className="c7"
        color="white"
        size="s"
      >
        Curabitur pretium tincidunt lacus. Nulla gravida orci a odio. Nullam varius, turpis et commodo, est eros bibendum elit, nec luctus magna sollicitudin.
      </p>
      <span>
        <a
          className="c8"
          color="white"
          href="/"
          target="_self"
          type="button"
        >
          Show me this
        </a>
      </span>
    </div>
  </div>
</div>
`;
